### ISSUE TITLE
Why are there no new releases?

### ISSUE THREAD

**Issue by sdan41517** [ytdl-org/youtube-dl#29753]
Is youtube-dl dead again? Are there no other maintainers that can approve the PRs?

Is there some legal issue going on in the background which has forced the development to be stopped? You guys can't just disappear on us like how you did before RIAA took down youtube-dl. At least add new maintainers that can take over the job while you guys are away. A lot of projects and software depend on youtube-dl.

**Comment by 89z** [comment]
Just use this:

~~~
pip install git+https://github.com/ytdl-org/youtube-dl
~~~

or, if you just need YouTube, I have an alternative here:

https://github.com/89z/mech

**Comment by rautamiekka** [comment]
If you need the commits that have come since the last release it helps to use the Git version with PIP, but otherwise it really doesn't ATM.

**Comment by sdan41517** [comment]
> Just use this:
> 
> ```
> pip install git+https://github.com/ytdl-org/youtube-dl
> ```
> 
> or, if you just need YouTube, I have an alternative here:
> 
> https://github.com/89z/mech

> If you need the commits that have come since the last release it helps to use the Git version with PIP, but otherwise it really doesn't ATM.

Is there a way to get the binary versions with the latest commits? Or do I need to build it myself?

**Comment by rautamiekka** [comment]
Need to build it yourself. You can cheat by using the "EXE" file that's made when installed through PIP, which is actually more like a self-extracting archive with a Python script inside that boots the actual Python source code installed alongside the "EXE".

**Comment by ugjka** [comment]
I've switched to yt-dlp

**Comment by 89z** [comment]
I created a new release with Windows build:

https://github.com/89z/mech/releases/tag/v1.24.0

However I cannot guarantee that will be done for every commit. I also added instruction here on how to build:

https://github.com/89z/mech/tree/master/cmd/youtube

**Comment by 89z** [comment]
I would ignore @fstirlitz, he seems to be a YT-DLP troll:

https://github.com/issues?q=involves:fstirlitz

**Comment by 89z** [comment]
My suspicion was confirmed, @fstirlitz just blocked me on GitHub. What a child.

**Comment by fstirlitz** [comment]
> My suspicion was confirmed, @fstirlitz just blocked me on GitHub. What a child.

@89z Because nothing screams ‘adult’ like calling people trolls completely unprovoked.

**Comment by 89z** [comment]
LOL @fstirlitz is sad. He unblocked me just to post a comment, then blocked again.

**Comment by 89z** [comment]
looks like we have another YT-DLP troll here @llacb47:

https://github.com/issues?q=involves:llacb47

**Comment by llacb47** [comment]
not a troll, but thank you for that link. Interesting to see how much time I've wasted on github lol @89z

**Comment by bnw42** [comment]
To echo the OP's question, is youtube-dl still alive? Two months is a long time between updates.

**Comment by 89z** [comment]
Please stop making pointless comments. Two months is not long in the terms of open source. I had an issue on another project (FFmpeg) that literally took five years to fix. If you dont like how long its been since a release, then use another project. I already made a suggestion above, plus other active good YouTube projects are available as well.

**Comment by filixm** [comment]
> Please stop making pointless comments. Two months is not long in the terms of open source. I had an issue on another project (FFmpeg) that literally took five years to fix. If you dont like how long its been since a release, then use another project. I already made a suggestion above, plus other active good YouTube projects are available as well. No one says you are required to use YouTube-DL and nothing else.

**Comment by filixm** [comment]
The question that was posed here is very valid.  I've been using youtube-dl for number of years, and this is the longest it has gone without an update, other than that time they were forced to shut down the repo due to RIAA lawsuit.  I did take your advise 89z and switched to a different repo, yt-dlp, thanks for the suggestion!

**Comment by bnw42** [comment]
Steven Penny wrote:
> Please stop making pointless comments. Two months is not long in the terms of open source. I had an issue on 
> another project (FFmpeg) that literally took five years to fix. If you dont like how long its been since a release,  then 
> use another project. I already made a suggestion above, plus other active good YouTube projects are available as well.

Given the large number of pending issues and that 2+ months is a long time for **_this_** project since the last update of any sort other than when it was taken down for legal reasons, I thought the OP's question, still not answered, was valid. There are plenty of users on other forums wondering what's happened to yt-dl viz updates. I'd already switched to another fork before the last update when yt-dl fell over with several streaming sites I frequent - they still don't work with yt-dl and issues were logged months ago .. I didn't get upset, I just moved to another fork.

For the person who is making the same criticising post quoted above & deleting it straight afterwards, this is not a whine about "why hasn't there been an update?" but rather a concerned "are there problems, legal or otherwise, behind the lengthy time since the last update?" It may take a long time (or never) to fix a specific issue, but we have gone over 2 months without a fix to *any* issue when previously updates came every few weeks. Users deserve to know whether a project is still alive or not. I've used other open source software where the devs have walked away yet the forum viz discussion & issues continued for years afterwards.

**Comment by fastily** [comment]
Definitely agree with the sentiments expressed above.  It does seem a bit odd that none of the maintainers (e.g. @dstftw, @remitamine) have been active on GitHub for the past couple months.  Yes, burnout is a very real issue, but I have to wonder why a successor was not named.  Based on the number of outstanding PRs, there seem to be plenty of eligible volunteers

**Comment by bnw42** [comment]
I've known of more than a few downloaders & front ends that weathered attacks from 'authorities' but suddenly went silent before completely vanishing after the said authorities switched from attacking the project to pursuing the developers. Yt-dl is a sizeable project with quite a few devs ... it seems odd that the whole project should suddenly freeze & the maintainers vanish .. it raises lots of red flags that there may have been some sort of legal action. The longer the uncertainty goes on, the more ominous it gets.

**Comment by ugjka** [comment]
Could it be because of "YouTube™" in the project's name?

**Comment by ugjka** [comment]
You have never met lawyers, have you?

**Comment by rautamiekka** [comment]
Not in person, but followed a couple, and a name as part of a software _hardly_ matters.

**Comment by bnw42** [comment]
youtube-dl may have originally started off many years ago as a youtube downloader, but it has long since branched out to work with almost any non-DRM'd streaming service. The "youtube" is merely a legacy of it's origins.

As for lawyers .. I've met some, some are even quite nice people. yt-dl has weathered legal attacks in the past and every time that's happened the news of the legal attack is not only mentioned on the yt-dl pages but it is reported in the technology media around the world. There's been no such reporting this time.

One reason why yt-dl has weathered previous legal attacks is that whilst it can be used for "breach of rights" downloads, it also has a lot of legitimate users. Educational institutions for instance, can obtain a licence to download streaming media for research and/or educational purposes.

**Comment by dimitarvp** [comment]
We are not entitled to the work of unpaid volunteers.

That being said, defensive or even flamey responses like the quotes above are not the way to go either. Putting a section on the top of the README, say named "Current status", that's being updated once every 1-2 weeks, is going to go a long way assuaging people's fears.

RIAA is a scary entity and it has some pretty scary people behind it as well. I think it's fair to expect that projects like `youtube-dl` should issue periodic "heartbeat" updates. Example: "We are still here and working hard but the current issues require more time. We'll post another update as we're progressing".

**Comment by bnw42** [comment]
> dimitarvp wrote ...

I fully agree. It's not just yt-dl though - the two co-ordinators for the yt-dl project have been missing in action from all of their github projects for 2 months now. And in their absence, yt-dl is in limbo. Unless the project co-ordinators are residents of the USA, the RIAA has much less power to threaten them than it would against github itself since Github is based in the USA and now owned by microsoft, so an easy target for the RIAA. Yet there is no sign, either on github (yt-dl has not been removed) or in the tech media of an attack on yt-dl. If the RIAA or some other entity had embarked on legal action against the yt-dl co-ordinators, part of that action would be a requirement to take down yt-dl. Since that has not happened, it strongly suggests their silence is due to other reasons & not legal attacks. Burnout & health reasons come to mind. :(

**Comment by rautamiekka** [comment]
Amen to that, exactly my thinking. COVID-19 is still in circulation, so that could end in multiple ways.

**Comment by furkanmustafa** [comment]
This is extremely concerning. I hope everything is all right.

I did not forget about Grooveshark yet. https://en.wikipedia.org/wiki/Josh_Greenberg#Death_and_legacy

**Comment by animelover1984** [comment]
The PRs have been piling up for years now; it isn't as though this is a recent development. I recommend using yt-dlp.

**Comment by dimitarvp** [comment]
I switched to `yt-dlp` today and was pleasantly surprised that it downloads even higher quality videos and, at times, a better audio stream as well. Also it does seem to handle YouTube's throttling very well.

Thanks for the recommendation! I'm very pleased with the switch.

**Comment by bnw42** [comment]
Re the reference to Grooveshark, when that takedown happened the service was no longer available and the takedown was widely reported in the media pretty quickly .. as with last year's takedown of yt-dl. There's been nothing this time, which suggests the issue is personal. I wasn't going to mention covid, but since someone else did, that was exactly what I was thinking. I lost a close friend to it a few months ago. :(

yt-dlp works a treat .. but be aware that quite a few of the command line options are different, including default options. Of course, most of the time the default settings for both are fine.

**Comment by artenaki** [comment]
NSA?

**Comment by rautamiekka** [comment]
Pfft, I highly doubt it. Like it was a national security problem.

**Comment by bnw42** [comment]
If it was any government or legal agency then the yt-dl project would have been taken down as well., After all, their goal would be to eradicate yd-dl, so going after devs would only be a means to that end. No, I'm more inclined towards health problems or burnout. Or they all decided to take a holiday at the same time.

**Comment by artenaki** [comment]
Two people at the same time, with no warning? Well, we'll see.

**Comment by bnw42** [comment]
We are in the middle of a pandemic, so more likely now than 2 years ago. :(   As for burnout, that can affect the whole team at the same time.

**Comment by nicolaasjan** [comment]
Burnout or not, they could have at least issued a statement.

**Comment by furkanmustafa** [comment]
Assuming that this is/was NOT the main thing in authors' lives.. burnout, or being too busy, or being engaged in an issue (like sickness), it's fairly possible to not comment at all, anywhere on the project. But all possible cases are concerning :(

Thanks for pointing out [yt-dlp](https://github.com/yt-dlp/yt-dlp), looks like that is already a replacement.

**Comment by myfirstnameispaul** [comment]
Looks like there were only a handful of commits in July and August of 2020. Maybe they are taking a break during these months?

**Comment by rusty-snake** [comment]
This issues can be closed IMHO, #30568.

**Comment by Lee-Carre** [comment]
> Why are there no new releases?

🙄🤦‍♂️

Due to global the supply-chain problems, we're awaiting shipment of necessities for the code-monkeys, in order that they consent to producing code. This being a 3-4 banana problem, after all. 😁
Plus, bananas are a (Cavendish) monoculture, so we must be extra careful that the plague doesn't start infecting them, which would require even more vaccine doses.

Of course, ample complaining by many users always greases the wheels.

You could always ask the code fairy, or Santa, instead. That would be quicker than installing a totalitarian government to whip keyboard-bashers into submission.

I hear that code-monkeys prefer free-range bananas, though.

**Comment by scrutinizer11** [comment]
It's another 7 months hiatus after the last update in Dec' 21. The oft-recommended **yt-dlp** failed to compile on my system and I'm forced to use **youtube-dl** on every site but _YouTube_. So the new update would be welcome, especially if  it manages to hammer out the throttling.

**Comment by nicolaasjan** [comment]
> It's another 7 months hiatus after the last update in Dec' 21. The oft-recommended **yt-dlp** failed to compile on my system and I'm forced to use **youtube-dl** on every site but _YouTube_. So the new update would be welcome, especially if it manages to hammer out the throttling.

If `yt-dlp` failed to compile on your macOS, open an issue at their repo.
If it's because your system Python version is still at **2.7**, you can download the macOS standalone build.
(see this [issue](https://github.com/yt-dlp/yt-dlp/issues/1075))

FYI, there was just released an [update](https://github.com/yt-dlp/yt-dlp/releases/tag/2022.06.22.1). 🙂


Also, when you'd compile `youtube-dl` now, you'll notice that the YouTube throttling issue is already fixed in the current `master`.

**Comment by scrutinizer11** [comment]
I installed Python 3. Building it still fails as 10.15SDK is expected (a configuration failure), while I'm on macOS 10.14.
Are you saying if I re-download the youtube-dl source and compile I could use it on YT as well? Or is it a static binary that you suggest?

**Comment by nicolaasjan** [comment]
> Building it still fails as 10.15SDK is expected (a configuration failure), while I'm on macOS 10.14.

I must admit that I know nothing about compiling on macOS, so I can't give you any advice regarding that. 😀️
(I build `youtube-dl` and `yt-dlp` on Linux Mint)

> Are you saying if I re-download the youtube-dl source and compile I could use it on YT as well?

That's what I do and it works on YouTube.

- `git clone https://github.com/ytdl-org/youtube-dl.git`
- `cd youtube-dl`
(then edit `./youtube_dl/version.py`)
- `make`

Then put the resulting static binary in `/usr/local/bin`.

But doesn't the [macOS standalone build](https://github.com/yt-dlp/yt-dlp/releases/download/2022.06.22.1/yt-dlp_macos) of `yt-dlp` work for you?

**Comment by myfirstnameispaul** [comment]
There are at least *17* subscribers to this issue and your troubleshooting seems off-topic.

Can you please create a new issue and resolve it there? It doesn't seem necessary to flood everyone's inbox with your discussion.
